### ISSUE TITLE
Cache not invalidated after changing parent 

### ISSUE THREAD

**Issue by eigan** [CuyZ/Valinor#131]
### Steps to reproduce
```php
<?php

// TopLevel.php

class TopLevel {
    /**
     * @var FooBar2
     */
    protected $foo;
}
``` 

```php
<?php

// Child.php

require "TopLevel.php";

class Child extends TopLevel {

    public int $bar;

    public string $title;
}

```

```php

// test.php

require "vendor/autoload.php";
require "Child.php";

$foo = (new \CuyZ\Valinor\MapperBuilder())
    ->mapper()
    ->map(Child::class, []);
```

1. Run `php test.php`
2. Error: The type `FooBar2` for property `TopLevel::$foo` could not be resolved: Cannot parse unknown symbol `FooBar2`
3. Now solve the issue, for instance by removing `$foo` from `TopLevel`
4. Run `php test.php` and the same error is thrown, as if the property wasn't removed.

**Comment by romm** [comment]
Hi @eigan thank you for the report, I can indeed see the issue here.

I already planned to change a bit how the cache is managed, I should be able to submit a patch quite soon: I'd be glad to have your feedback on it so I'll keep you updated here.

**Comment by romm** [comment]
Hi @eigan, the changes in #132 should fix this issue. Would you mind checking the PR branch out and testing locally if the issue is fixed?

Thanks!

**Comment by eigan** [comment]
@romm Branch `feat/file-system-cache` with configured `FileWatchingCache` does not have this issue 👍🏻

**Comment by romm** [comment]
Closed with #132.
